sample-dapps/ethereum-wallet-explorer/src/components/explorer/tokens/TokensTable.tsx

```diff
@@ -0,0 +1,97 @@
+"use client"
+import React, { useEffect, useState } from 'react';
+import { TableCell, TableRow, TableBody, Table, TableHeader, TableHead } from "../../ui/table";
+import { Button } from '../../ui/button';
+import TableSkeleton from '../helpers/TableSkeleton';
+
+interface ERC20Token {
+    name: string;
+    symbol: string;
+    decimals: string;
+    address: string;
+    quantityIn: string;
+    quantityOut: string;
+    totalBalance: string;
+}
+
+interface TokenProps {
+    walletAddress: string;
+}
+
+const TokensTable = ({ walletAddress }: TokenProps) => {
+    const [tokens, setTokens] = useState<ERC20Token[]>([]);
+    const [totalPages, setTotalPages] = useState(0);
+    const [currentPage, setCurrentPage] = useState(1);
+    const [totalTokens, setTotalTokens] = useState(0);
+    const [loading, setLoading] = useState(true);
```

Comment:
same feedback as above on default states. if you change anything in other component, do here as well